### ISSUE TITLE
Set time between runs

### ISSUE THREAD

**Issue by boriseetgerink** [fluentscheduler/FluentScheduler#112]
I have a job that runs continuously, can take a variable amount of time to complete and I want to wait a certain amount of time between its completion and before it starts again. Also, I don't want it to run in parallel. Is this possible with FluentScheduler?

For example: I want to have a 10 second delay between the completion and restart of a job. If I set it to run every 10 seconds and the execution takes 9 seconds than the job would only 'wait' 1 second before starting again, not the desired 10 seconds.

I know I can make the job non-reentrant, but that's only part of the solution. Perhaps the NextRun property can be calculated after the job completes?

**Comment by tallesl** [comment]
Unfortunately, you can't do that in the current version of the library.

You can set it to be reentrant and you can also set a delay before it runs, but you're unable to set such delay to be calculated accordingly to the last run.

It would be a lovely feature, but that's not the easiest thing in the world to implement.
The main problem with it is making sure the time (even if it' very few milliseconds) that the library spends 'with itself' doesn't interfere in the calculation.

I'm letting it rest here as 'new feature' but, being honest with you, this have a lower priority with me compared to the other open issues.

**Comment by boriseetgerink** [comment]
Thanks for the clarification. Considering the requirements I can see it would be hard to implement.
